src/task.py

```diff
@@ -1,59 +1,61 @@
-import variable
 import os
 import re
 
+import variable
+
 class IndexedFile:
     def __init__(self, index, name):
         self.index = index
         self.name = name
 
-class InputMiss(Exception):
-    def __init__(self, expect_file):
-        super().__init__(f'Input file for {expect_file} is missing')
+    def __lt__(self, other):
+        return self.index < other.index
 
-class ExpectMiss(Exception):
-    def __init__(self, input_file):
-        super().__init__(f'Expect file for {input_file} is missing')
+class FileMiss(Exception):
+    def __init__(self, format_type, index):
+        super().__init__(f'File in "{format_type}" type with index "{index}" is missing')
 
-def expect_file(input_file, expect_hash):
-    if input_file.index in expect_hash:
-        # FIXME: the function pop() is not efficient since it will modify the hash table
-        return expect_hash.pop(input_file.index).name
-    else:
-        raise ExpectMiss(input_file.name)
+def build_tasks(files_info):
+    keys = list(files_info.keys())
+    files = {}
+    indices = []
 
-def build_tasks(input_list, expect_list):
-    expect_hash = {file.index: file for file in expect_list}
+    for key in keys:
+        files[key] = [file.name for file in files_info[key]]
+    if len(keys) > 0:
+        indices = [file.index for file in files_info[keys[0]]]
 
-    tasks = [(input.name, expect_file(input, expect_hash)) for input in input_list]
+    return (files, indices)
```

Comment:
My consideration is that
- May not return `IndexedFile` type data.
- May return the name of each type of files, such as "input", "expect", and so on. (which is not returned in `main` branch code)

So the changed return value format may seem like the following part.

```yaml
# (files, indices)
(
    {
        "input": ["input_1.txt", ......],
        "expect": ["expect_1.txt", ......]
    },
    ["1", ......] 
}
```

I agree the part that we should coupling "input" and "expect" together, so I may change the format like below.

```yaml
# files
{
    "input": ["input_1.txt", ......],
    "expect": ["expect_1.txt", ......],
    "index": ["1", ......] 
}
```

Reply:
First of all,
- I agree that returning `IndexedFile` isn't a good idea.

In my opinion, about the returning structure, the following  might better.
```yaml
[
    (
        (1), 
        {
            "input": "input_1.txt",
            "expect": "expect_1.txt",
        }
    ),
    ...
]
```
or
```yaml
[
    (
        {
           "index": (1),
            "input": "input_1.txt",
            "expect": "expect_1.txt",
        }
    ),
    ...
]
```

In these case, we can process the returned tasks in the following way
```python3
tasks = task.list_tasks(config)

# Process tasks
test_result = {}
for task in tasks:
    for format, file in task.items():
        config[format] = file
    (log, result) = execute_commands(config, 'process')
```
which I thinks is more readable.

Reply:
I use the first structure at last. Because `index` and `files` are different kind of data, I prefer to split them into different part. So finally, I handle `tasks` in `process.py` like below.

```python
tasks = task.list_tasks(config)

# Process tasks
test_result = {}
for (index, files) in tasks:
    for file_type, file_path in files.items():
        config[file_type] = file_path
    (log, result) = execute_commands(config, 'process')
```

For reference, the structure is

```yaml
[
    (
        (1), 
        {
            "input": "input_1.txt",
            "expect": "expect_1.txt",
        }
    ),
    ...
]
```